Cargo.toml

```diff
@@ -329,6 +329,7 @@ reed-solomon-erasure = "6.0.0"
 regex = "1.11.1"
 reqwest = { version = "0.11.27", default-features = false }
 reqwest-middleware = "0.2.5"
+rocksdb = { version = "0.23.0", default-features = false }
```

Comment:
Can you elaborate on why we want this change ? Currently, `solana-ledger` is the only crate that directly depends on `rocksdb` and one could argue that having the definition in `solana-ledger` crates keeps the info "closer" to where it applies. I'm not positive if we have a "rule" about which one to prefer in our repo

One item I do see is that if we had a second crate start depending on rocksdb, we'd avoid a duplicate at that point. This seems somewhat unlikely to me at this point